Relax mayDependOnBaseURL condition for custom property

Constellation opened the pull request:
#### 8930049ceecd1c421783172cc234a0cbe3eab94d
<pre>
Relax mayDependOnBaseURL condition for custom property
<a href="https://bugs.webkit.org/show_bug.cgi?id=275602">https://bugs.webkit.org/show_bug.cgi?id=275602</a>
<a href="https://rdar.apple.com/129291831">rdar://129291831</a>

Reviewed by Antti Koivisto.

Before 275795@main, our caching inline style cache had a bug that it cannot handle base URL resolution correctly.
It had many bugs for nested CSSValues, missing CSS properties and so on, and 275795@main fixed them.
For CSSCustomPropertyValue, we disabled inline style cache conservatively since it may include some relative URL
resolution with base URL (previously, we completely ignored and shared the cache. That was completely wrong.)
But it turned out that our bug caused some caching works, and now they are disabled because of the fix.
In this patch, we add a bit more precise handling of CSSCustomPropertyValue for mayDependOnBaseURL information to
expand caching effectiveness.

* Source/WebCore/css/CSSCustomPropertyValue.cpp:
(WebCore::mayDependOnBaseURL):
(WebCore::CSSCustomPropertyValue::customMayDependOnBaseURL const):
* Source/WebCore/css/CSSCustomPropertyValue.h:

Canonical link: <a href="https://commits.webkit.org/280125@main">https://commits.webkit.org/280125@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9b44a3c5fedf10f0cc254395f1c6a6eee85256f6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55807 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35131 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8275 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58796 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6239 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57933 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42752 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6435 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/44940 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/4304 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57835 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/33029 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/48106 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26073 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/29813 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/5434 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4382 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/51794 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/5705 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60389 "Built successfully") | 
| | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-1-Build-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/5836 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/52377 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🛠 vision-sim ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/48175 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/51870 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8234 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/30960 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32044 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33126 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/31792 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

mdubet commented on Source/WebCore/css/CSSCustomPropertyValue.cpp:
```diff
@@ -192,4 +192,54 @@ bool CSSCustomPropertyValue::isAnimatable() const
+static bool mayDependOnBaseURL(const CSSCustomPropertyValue::SyntaxValue& syntaxValue)
+{
+    return WTF::switchOn(syntaxValue,
+        [&](const Length&) {
```
Nit: We don't need all the lambda `&` capture here

Constellation replied:
Done

Constellation committed:
Relax mayDependOnBaseURL condition for custom property
https://bugs.webkit.org/show_bug.cgi?id=275602
rdar://129291831

Reviewed by Antti Koivisto.

Before 275795@main, our caching inline style cache had a bug that it cannot handle base URL resolution correctly.
It had many bugs for nested CSSValues, missing CSS properties and so on, and 275795@main fixed them.
For CSSCustomPropertyValue, we disabled inline style cache conservatively since it may include some relative URL
resolution with base URL (previously, we completely ignored and shared the cache. That was completely wrong.)
But it turned out that our bug caused some caching works, and now they are disabled because of the fix.
In this patch, we add a bit more precise handling of CSSCustomPropertyValue for mayDependOnBaseURL information to
expand caching effectiveness.

* Source/WebCore/css/CSSCustomPropertyValue.cpp:
(WebCore::mayDependOnBaseURL):
(WebCore::CSSCustomPropertyValue::customMayDependOnBaseURL const):
* Source/WebCore/css/CSSCustomPropertyValue.h:

Canonical link: https://commits.webkit.org/280125@main
